diff --git a/src/server/game/Entities/Player/Player.h b/src/server/game/Entities/Player/Player.h
index 774e751..567ba99 100755
--- a/src/server/game/Entities/Player/Player.h
+++ b/src/server/game/Entities/Player/Player.h
@@ -823,7 +823,8 @@ enum PlayerDelayedOperations
 
 // Player summoning auto-decline time (in secs)
 #define MAX_PLAYER_SUMMON_DELAY                   (2*MINUTE)
-#define MAX_MONEY_AMOUNT                       (0x7FFFFFFF-1)
+#define MAX_MONEY_AMOUNT                        (0xFFFFFFFF) // uint32 max
+#define OLD_MAX_MONEY_AMOUNT                    (0x7FFFFFFF-1) // using to error out things that wont work etc (int32 max -1)
 
 struct InstancePlayerBind
 {
diff --git a/src/server/game/Handlers/MailHandler.cpp b/src/server/game/Handlers/MailHandler.cpp
index 69a8489..4795611 100755
--- a/src/server/game/Handlers/MailHandler.cpp
+++ b/src/server/game/Handlers/MailHandler.cpp
@@ -113,6 +113,12 @@ void WorldSession::HandleSendMail(WorldPacket & recv_data)
         return;
     }
 
+    if (reqmoney > OLD_MAX_MONEY_AMOUNT)
+    {
+        player->SendMailResult(0, MAIL_SEND, MAIL_ERR_INTERNAL_ERROR);
+        return;
+    }
+
     Player* receive = ObjectAccessor::FindPlayer(rc);
 
     uint32 rc_team = 0;
@@ -445,6 +451,12 @@ void WorldSession::HandleMailTakeItem(WorldPacket & recv_data)
         return;
     }
 
+    if (m->COD > OLD_MAX_MONEY_AMOUNT)
+    {
+        player->SendMailResult(mailId, MAIL_ITEM_TAKEN, MAIL_ERR_INTERNAL_ERROR);
+        return;
+    }
+
     Item* it = player->GetMItem(itemId);
 
     ItemPosCountVec dest;